### ISSUE TITLE
Port scan failed

### ISSUE THREAD

**Issue by bbkali** [yogeshojha/rengine#830]
Customized scanning engine, as long as there is port scanning, an error will be reported and displayed

Error: Field 'number' expected a number but got {'Port': 80, 'Protocol': 0, 'TLS': False}.
Vulnerability ScanIn progress

The port file was also not found
<img width="1186" alt="image" src="https://user-images.githubusercontent.com/25882066/222658064-ade6b5dc-6dc7-446d-b3a3-f261ba432714.png">

<img width="318" alt="image" src="https://user-images.githubusercontent.com/25882066/222657891-e654019e-661f-47ad-a3c4-534271432a80.png">

**Comment by bbkali** [comment]
> 有同样的问题。#674和#689中提到的问题尝试了解决方案并重新安装了很多次，但没有成功。

没有解决这个问题的方法，我使用的是 1.3.6 版，系统是kali 2022 但问题仍然存在,使用的是自定义扫描引擎,只要选择端口扫描就会报错"Scan Failed due to: Field 'number' expected a number but got {'Port': 80, 'Protocol': 0, 'TLS': False}."

**Comment by bbkali** [comment]
there is no way to solve this problem. I use version 1.3.6, but the problem still exists 
"Scan Failed due to: Field 'number' expected a number but got {'Port': 80, 'Protocol': 0, 'TLS': False}."

**Comment by zinwelzl** [comment]
Same here, v1.3.6, "Port Scanning Failed - Error: Field 'number' expected a number but got {'Port': 80, 'Protocol': 0, 'TLS': False}."
Try on ubuntu server, kali and parrot.

**Comment by geekjy** [comment]
same,hope it can be solved quickly

**Comment by Dody2004Sec** [comment]
same here

**Comment by sxyrxyy** [comment]
to anyone having one of those issues:
1) Scan Failed due to: [Errno 2] No such file or directory: '/usr/src/scan_results/domain.com_304237806434/ports.json'
2) Scan Failed due to: Field 'number' expected a number but got {'Port': 80, 'Protocol': 0, 'TLS': False}.

1) Make sure "use_naabu_config" is set to false within the YAML
   The script tried to call the naabu binary with the "-config" argument which is not existent.

2) Seems like a parsing error, the "port_number" variable needs to be changed.
   port_number = json_st['port'] ---> port_number = json_st['port']['Port']

if the "ports.json" problem persists, it might be that naabu didnt find anything.

**Comment by dontclickthelink** [comment]
Naabu changed how it reports the port scan results. Making a simple change to the tasks.py file fixes the problem in my testing. Check out #833 for details.

**Comment by bbkali** [comment]
During use, the following problems were found

Scan Failed due to: '>' not supported between instances of 'str' and 'int'

Is it because the port scan failed?

**Comment by dontclickthelink** [comment]
> During use, the following problems were found
> 
> Scan Failed due to: '>' not supported between instances of 'str' and 'int'
> 
> Is it because the port scan failed?

Honestly, I'm not sure. I haven't encountered that error message (yet). After the change (above) to the tasks.py file, all my scans are running flawlessly. I'll take a deeper look at your error in a bit.

**Comment by bbkali** [comment]
I thought it was a problem with my scan target, and now I have tried to scan multiple different targets, all of which are in this error

Scan Failed due to: '>' not supported between instances of 'str' and 'int'

This seems to be an error in the Directory Search module. I checked task.py. Is there a problem of inconsistent types in the comparison of the > symbol? How to modify it?
<img width="337" alt="image" src="https://user-images.githubusercontent.com/25882066/223623841-d74c6659-85f0-45e8-8026-17b7c508bf06.png">
<img width="980" alt="image" src="https://user-images.githubusercontent.com/25882066/223623902-9e690651-5d44-40ac-9175-4a6ad15b9c64.png">
